lib/android-tools-info.ts

```diff
@@ -0,0 +1,209 @@
+///<reference path=".d.ts"/>
+"use strict";
+
+import * as path from "path";
+import * as semver from "semver";
+
+export class AndroidToolsInfo implements IAndroidToolsInfo {
+	private static ANDROID_TARGET_PREFIX = "android";
+	private static SUPPORTED_TARGETS = ["android-17", "android-18", "android-19", "android-21", "android-22"];
+	private static MIN_REQUIRED_COMPILE_TARGET = 21;
+	private static REQUIRED_BUILD_TOOLS_RANGE_PREFIX = ">=22";
+	private static VERSION_REGEX = /^(\d+\.){2}\d+$/;
+	private showWarningsAsErrors: boolean;
+	private toolsInfo: IAndroidToolsInfoData;
+	private selectedCompileSdk: number;
+	private installedTargetsCache: string[] = null;
+
+	constructor(private $childProcess: IChildProcess,
+		private $errors: IErrors,
+		private $fs: IFileSystem,
+		private $logger: ILogger,
+		private $options: IOptions) {}
+
+	public getToolsInfo(options?: {showWarningsAsErrors: boolean}): IFuture<IAndroidToolsInfoData> {
+		return ((): IAndroidToolsInfoData => {
+			if(!this.toolsInfo) {
+				this.showWarningsAsErrors = options && options.showWarningsAsErrors;
+
+				let infoData: IAndroidToolsInfoData = Object.create(null);
+				infoData.androidHomeEnvVar = this.getAndroidHome();
+				infoData.compileSdkVersion = this.getCompileSdk().wait();
+				infoData.buildToolsVersion = this.getBuildToolsVersion().wait();
+				infoData.targetSdkVersion = this.getTargetSdk().wait();
+				infoData.supportLibraryVersion = this.getAndroidSupportLibVersion().wait();
+
+				this.toolsInfo = infoData;
+			}
+
+			return this.toolsInfo;
+		}).future<IAndroidToolsInfoData>()();
+	}
+
+	public validateInfo(options?: {showWarningsAsErrors: boolean, validateTargetSdk: boolean}): IFuture<void> {
+		return (() => {
+			let toolsInfoData = this.getToolsInfo({showWarningsAsErrors: options && options.showWarningsAsErrors}).wait();
+			if(!toolsInfoData.androidHomeEnvVar || !this.$fs.exists(toolsInfoData.androidHomeEnvVar).wait()) {
+				this.printMessage("ANDROID_HOME environment variable is not set or it points to not existing directory. You cannot use gradle without setting it.",
```

Comment:
The ANDROID_HOME environment variable is not set or it points to a non-existent directory. You will not be able to perform any build-related operations for Android.